src/FAudio_internal.c

```diff
@@ -491,19 +491,24 @@ static void FAudio_INTERNAL_DecodeBuffers(
 				/* Callbacks */
 				if (voice->src.callback != NULL)
 				{
-					if (voice->src.callback->OnBufferEnd != NULL)
```

Comment:
Now only call one call back or the other based on whether there are more samples to send.

Reply:
This violates the spec and has to be undone.

Reply:
Unit test to verify https://github.com/FNA-XNA/FAudio/commit/14a44cadac8cc685645f12920be05f6f8fd70f50